controllers/managedocs_controller_test.go

```diff
@@ -1349,7 +1349,14 @@ var _ = Describe("ManagedOCS controller", func() {
 				sm.Spec = promv1.ServiceMonitorSpec{
 					Endpoints: []promv1.Endpoint{},
 				}
-				Expect(k8sClient.Update(ctx, sm)).Should(Succeed())
+
+				// Contention of test and reconilcer updating the resource is observed
+				// try to get the update from test succeed before polling for resource change
+				Eventually(func() bool {
+					err := k8sClient.Update(ctx, sm)
+					return err == nil
+				}, timeout, interval).Should(BeTrue())
```

Comment:
I would prefer to have to use `consistently` on the get to validate a stable resource version.

Reply:
makes sense, done.